### PR TITLE
P2022-1380 Build board columns view pager based on configuration -> P2022-1645 Connect with API

### PR DESCRIPTION
**Pull request by Nataniel-Antosik**

### CODE REVIEW

**Comment by Nataniel-Antosik** [comment]
I have a problem with connecting API. I see a 'loading' message and 'something went wrong but I didn't see a problem in logcat. When I sent to UI mock data I didn't have any problem with an adapter. I thinking about the wrong data name in the model but I change it by 'SerializedName'.
Below is a screen from logcat.
Zrzut ekranu 2022-03-16 142807
And video from app.

https://user-images.githubusercontent.com/57491794/158601764-c5528003-fd11-42a2-8d5e-435d19fec68b.mp4

File: app/src/main/res/layout/retro_fragment.xml [hunk 1]
```diff
@@ -1,22 +1,47 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+        <LinearLayout
+        android:id="@+id/linearLayout"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:orientation="vertical">
```
Comment (MarekMacko) [inline app/src/main/res/layout/retro_fragment.xml:26]: Is this `LinearLayout` used anywhere? Seems to be redundant

File: app/src/main/res/layout/retro_fragment.xml [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
+            app:tabMode="scrollable" />
+
+        <androidx.viewpager2.widget.ViewPager2
+            android:id="@+id/view_pager_retro"
+            android:layout_width="match_parent"
+            android:layout_height="0dp"
+            android:layout_weight="1" >
+
+
+        </androidx.viewpager2.widget.ViewPager2>
```
Comment (MarekMacko) [inline app/src/main/res/layout/retro_fragment.xml:43]: There is no need for a closing tag if there is no children in. 
You remove `</androidx.viewpager2.widget.ViewPager2>` and use a closing expression `/>` - like for `TabLayout`

File: app/src/main/java/com/intive/patronage/retro/retro/presentation/entity/Retro.kt [hunk 3]
```diff
@@ -2,4 +2,4 @@ package com.intive.patronage.retro.retro.presentation.entity
 
 import com.intive.patronage.retro.board.presentation.entity.Board
 
-data class Retro(val board: Board, val listColumns: List<Columns>, val listUsers: List<User>)
+data class Retro(val board: Board, val columns: List<Columns>, val users: List<User>)
```
Comment (MarekMacko) [inline app/src/main/java/com/intive/patronage/retro/retro/presentation/entity/Retro.kt:5]: Next time would be nice to separate a renaming of fields to a separate PR

File: app/src/main/java/com/intive/patronage/retro/retro/presentation/view/RetroFragment.kt [hunk 4]
```diff
@@ -1,59 +1,76 @@
...
 
         return binding.root
     }
 
     private fun setViewPager() {
-        val dataForTabName = retroViewModel.takeRetroConfiguration().flatMap { it.listColumns }
 
-        viewPager = binding.viewPagerRetro
-        tab = binding.tabLayoutRetro
+        token.observe(viewLifecycleOwner) { isTokenGenerated ->
```
Comment (MarekMacko) [inline app/src/main/java/com/intive/patronage/retro/retro/presentation/view/RetroFragment.kt:50]: Is this waiting is mandatory here? Wouldn't be the case that token is loaded before we reach `RetroFragment` screen?

File: app/src/main/res/layout/retro_fragment.xml (hunk 1 above)
Comment (Nataniel-Antosik) [reply app/src/main/res/layout/retro_fragment.xml:26]: I used it to configure tablayout and viewpager 2 together, and constraintlayout I used to show progresssindicator in the middle of app.

File: app/src/main/res/layout/retro_fragment.xml (hunk 2 above)
Comment (Nataniel-Antosik) [reply app/src/main/res/layout/retro_fragment.xml]: I changed it.

File: app/src/main/java/com/intive/patronage/retro/retro/presentation/view/RetroFragment.kt (hunk 4 above)
Comment (Nataniel-Antosik) [reply app/src/main/java/com/intive/patronage/retro/retro/presentation/view/RetroFragment.kt]: No is not mandatory here, I deleted it.

File: app/src/main/java/com/intive/patronage/retro/retro/presentation/entity/Retro.kt (hunk 3 above)
Comment (Nataniel-Antosik) [reply app/src/main/java/com/intive/patronage/retro/retro/presentation/entity/Retro.kt]: I'll remember it well, now I took it back and used SerializedName

File: app/src/main/res/layout/retro_fragment.xml (hunk 1 above)
Comment (MarekMacko) [reply app/src/main/res/layout/retro_fragment.xml:26]: If you're using `ConstraintLayout` then try to forget about old-fashioned ViewGroup layouts. 
- top of TabLayout should be constrained to the top of parent,
- top of ViewPager should be constrained to the bottom of TabLayout,
- bottom of ViewPager should be constrained to the bottom of parent.

**Review by kmaslowiec — APPROVED** [review]
Good
